bootstore/src/schemes/v0/messages.rs

```diff
@@ -0,0 +1,130 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Messages sent between peers
+
+use crate::trust_quorum::{LearnedSharePkgV0, SharePkgV0};
+use derive_more::From;
+use serde::{Deserialize, Serialize};
+use sled_hardware::Baseboard;
+use uuid::Uuid;
+
+/// The first thing a peer does after connecting or accepting is to identify
+/// themselves to the connected peer.
+///
+/// This message is interpreted at the peer (network) level, and not at the FSM level,
+/// because it is used to associate IP addresses with [`Baseboard`]s.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Identify(Baseboard);
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Envelope {
+    pub to: Baseboard,
+    pub msg: Msg,
+}
+
+#[derive(From, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Msg {
+    Req(Request),
+    Rsp(Response),
+}
+
+/// A request sent to a peer
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Request {
+    // A counter to uniquely match a request to a response for a given peer
+    pub id: Uuid,
+    pub type_: RequestType,
+}
+
+/// A response sent from a peer that matches a request with the same sequence
+/// number
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Response {
+    pub request_id: Uuid,
+    pub type_: ResponseType,
+}
+
+/// A request from a peer to another peer over TCP
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum RequestType {
+    /// A rack initialization request informing the peer that it is a member of
+    /// the initial trust quorum.
+    Init(SharePkgV0),
```

Comment:
Would it make sense to move this type into the `v0` module, instead of versioning it in the name?

Reply:
Done in [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98)